Implement LocalAccount equality test

## What was wrong?

### Code that produced the error

```py
from eth_account import Account
key = b'a' * 32
acct1 = Account.privateKeyToAccount(key)
acct2 = Account.privateKeyToAccount(key)
assert acct1 == acct2, "local accounts with the same private key should evaluate as equal"
```

### Full error output 

```sh
AssertionError: local accounts with the same private key should evaluate as equal
```

## How can it be fixed?

Add an `__eq__` to `LocalAccount`, test that both objects are exactly the same type, and that they have the same key.

Bonus:  Make the private key attribute of `LocalAccount` immutable, and implement `__hash__` as `hash(<private_key>)`.

--- a/eth_account/local.py
+++ b/eth_account/local.py
@@ -29,9 +29,16 @@ class LocalAccount(object):
         self.address = key.public_key.to_checksum_address()
 
         key_raw = key.to_bytes()
-        self.privateKey = key_raw
+        self._privateKey = key_raw
 
         self._key_obj = key
+
+    @property
+    def privateKey(self):
+        '''
+        Get the private key.
+        '''
+        return self._privateKey
 
     def encrypt(self, password):
         '''
@@ -59,3 +66,9 @@ class LocalAccount(object):
 
     def __bytes__(self):
         return self.privateKey
+
+    def __hash__(self):
+        return hash(self.privateKey)
+
+    def __eq__(self, other):
+        return (isinstance(other, type(self))) and (self.privateKey == other.privateKey)

--- a/tests/core/test_accounts.py
+++ b/tests/core/test_accounts.py
@@ -82,6 +82,13 @@ def test_eth_account_create_variation(acct):
     assert account1 != account2
 
 
+def test_eth_account_equality(acct):
+    key = b'a' * 32
+    acct1 = acct.privateKeyToAccount(key)
+    acct2 = acct.privateKeyToAccount(key)
+    assert acct1 == acct2
+
+
 def test_eth_account_privateKeyToAccount_reproducible(acct, PRIVATE_BYTES):
     account1 = acct.privateKeyToAccount(PRIVATE_BYTES)
     account2 = acct.privateKeyToAccount(PRIVATE_BYTES)

